### ISSUE TITLE
does not work on YYYY-MM-DD

### ISSUE THREAD

**Issue by angelxmoreno** [angelxmoreno/axios-date-transformer#11]
the regex seems to only match full date strings (YYYY-MM-DDTHH:mm:ss) and not dates (YYYY-MM-DD). 

this could lead to breaking changes for users that found this to be expected. That being said, we should add a flag to trea date strings as Date objects.

**Comment by angelxmoreno** [comment]
this regex seems a bit more inclusive:

```ts
const dateRegex = /^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(?:\.\d*)?(?:[-+]\d{2}:?\d{2}|Z)?)?$/;
```

**Comment by OlliL** [comment]
You'll always end up with a solution where you can't guarantee that someone indeed wanted to have '2023-12-31' as a string instead of a date because its for example just a reference to whatever. The root cause is, that the response transformers have no idea about the object declaration they are about to create. So they never will have the chance to check the expected target type of each key but have to guess based on data. This will always lead to produce errors - unfortunately - with user input data.
I'm also eagerly looking for a clean solution to that problem - but I guess none exists unfortunately.

**Comment by angelxmoreno** [comment]
To clarify, what @OlliL is referring to is an example where a user chooses to name themself `1980-01-01`. when a json with a key containing their name is parsed, the name field which is expected to be a string is now a Date object.

This is indeed a real issue @OlliL . I will made a separate issue to try to address it #12

**Comment by angelxmoreno** [comment]
released in v1.10.0 as a minor instead of a patch since existing users would now have date strings converted to date objects
